### PR TITLE
Date time picker

### PR DESCRIPTION
**Pull request by anshul2401**

### CODE REVIEW

File: lib/screens/carrier/add_edit.dart [hunk 1]
```diff
@@ -49,46 +69,64 @@ class _CarrierAddEditState extends State<CarrierAddEdit> {
...
+                    validator: (val) => val.trim().isEmpty
+                        ? "Carrier name should not be empty"
+                        : null,
+                    onSaved: (value) => _carrierName = value,
+                  ),
+                  SizedBox(
+                    height: 10,
+                  ),
+                  GestureDetector(
+                    child: Text(
```
Comment (harsimranmaan) [inline lib/screens/carrier/add_edit.dart:87]: Use an IconButton

File: lib/screens/carrier/add_edit.dart [hunk 2]
```diff
@@ -30,9 +31,28 @@ class _CarrierAddEditState extends State<CarrierAddEdit> {
   String _carrierName;
   CarrierType _carrierType;
   Tenant _tenant;
+  DateTime pickedDate;
+  final DateFormat formatter = DateFormat('dd-MM-yyyy');
 
   @override
   Widget build(BuildContext context) {
+    void _presentDatePicker() {
+      showDatePicker(
```
Comment (harsimranmaan) [inline lib/screens/carrier/add_edit.dart:40]: Use a Datetimepicker

File: lib/screens/carrier/add_edit.dart [hunk 3]
```diff
@@ -30,9 +31,28 @@ class _CarrierAddEditState extends State<CarrierAddEdit> {
...
+              context: context,
+              initialDate: DateTime.now(),
+              firstDate: DateTime(2019),
+              lastDate: DateTime(2030))
```
Comment (harsimranmaan) [inline lib/screens/carrier/add_edit.dart:44]:
```suggestion
              lastDate: DateTime(2499)
```

File: lib/screens/carrier/add_edit.dart [hunk 4]
```diff
@@ -49,46 +69,64 @@ class _CarrierAddEditState extends State<CarrierAddEdit> {
...
-                          : null,
-                      onSaved: (value) => _carrierName = value,
+            children: [
+                  TextFormField(
+                    initialValue: widget.carrier.name ?? '',
+                    autofocus: widget.isEdit ? false : true,
+                    style: Theme.of(context).textTheme.headline5,
+                    decoration: InputDecoration(hintText: 'Carrier Name'),
+                    validator: (val) => val.trim().isEmpty
+                        ? "Carrier name should not be empty"
```
Comment (harsimranmaan) [inline lib/screens/carrier/add_edit.dart:79]: Can we move these to localization?

File: lib/screens/carrier/add_edit.dart [hunk 5]
```diff
@@ -30,9 +32,21 @@ class _CarrierAddEditState extends State<CarrierAddEdit> {
   String _carrierName;
   CarrierType _carrierType;
   Tenant _tenant;
+  var eta;
```
Comment (harsimranmaan) [inline lib/screens/carrier/add_edit.dart:35]:
```suggestion
  var _eta;
```

**Comment by anshul2401** [comment]
Screenshot_1606412675
Screenshot_1606412717

**Review by harsimranmaan — APPROVED** [review]

**Pull request merged**